diff -rupN emscripten-fastcomp-1.22.1/lib/ExecutionEngine/JIT/JITMemoryManager.cpp emscripten-fastcomp-1.22.1b/lib/ExecutionEngine/JIT/JITMemoryManager.cpp
--- emscripten-fastcomp-1.22.1/lib/ExecutionEngine/JIT/JITMemoryManager.cpp	2014-08-07 16:38:04.000000000 -0500
+++ emscripten-fastcomp-1.22.1b/lib/ExecutionEngine/JIT/JITMemoryManager.cpp	2014-08-21 16:34:11.924772000 -0500
@@ -826,12 +826,14 @@ public:
     sys::DynamicLibrary::AddSymbol("stat", (void*)(intptr_t)stat);
     sys::DynamicLibrary::AddSymbol("fstat", (void*)(intptr_t)fstat);
     sys::DynamicLibrary::AddSymbol("lstat", (void*)(intptr_t)lstat);
+#ifndef __ANDROID__
     sys::DynamicLibrary::AddSymbol("stat64", (void*)(intptr_t)stat64);
     sys::DynamicLibrary::AddSymbol("\x1stat64", (void*)(intptr_t)stat64);
     sys::DynamicLibrary::AddSymbol("\x1open64", (void*)(intptr_t)open64);
     sys::DynamicLibrary::AddSymbol("\x1lseek64", (void*)(intptr_t)lseek64);
     sys::DynamicLibrary::AddSymbol("fstat64", (void*)(intptr_t)fstat64);
     sys::DynamicLibrary::AddSymbol("lstat64", (void*)(intptr_t)lstat64);
+#endif
     sys::DynamicLibrary::AddSymbol("atexit", (void*)(intptr_t)atexit);
     sys::DynamicLibrary::AddSymbol("mknod", (void*)(intptr_t)mknod);
   }
diff -rupN emscripten-fastcomp-1.22.1/lib/Support/Unix/Process.inc emscripten-fastcomp-1.22.1b/lib/Support/Unix/Process.inc
--- emscripten-fastcomp-1.22.1/lib/Support/Unix/Process.inc	2014-08-07 16:38:04.000000000 -0500
+++ emscripten-fastcomp-1.22.1b/lib/Support/Unix/Process.inc	2014-08-21 14:39:05.032330000 -0500
@@ -36,7 +36,13 @@
 #  include <termios.h>
 #endif
 
+#ifndef __ANDROID__
 #include <sys/unistd.h>
+#endif
+
+#ifdef __ANDROID__
+#include <unistd.h>
+#endif
 
 //===----------------------------------------------------------------------===//
 //=== WARNING: Implementation here must contain only generic UNIX code that
